https/gifts.js

```diff
@@ -16,140 +16,242 @@ const { EVENT } = require("../config/constants");
 const db = require("../config/db");
 
 /**
- * @namespace sendGift
- * @return {sendGift~inner} - the returned function
+ * @namespace updateGiftSentStatus
+ * @return {updateGiftSentStatus~inner} - the returned function
  */
-const sendGift = functions.https.onCall(
+const updateGiftSentStatus = functions.https.onCall(
   /**
-   * marks the gift sent on boith the gifter and giftees account
+   * Updates the sent status of the gift
    * @inner
    * @param {object} data - details about the giftee
    * @param {string} data.user - user object or uid
-   * @param {boolean} data.value - marks it either true or false
-   * @param {string} data.gifteeGameAccountUUID - UUID of the giftee, if you do not know it
-   * @param {object} [context] - This is used by firebase, no idea what it does, I think its added automatically
+   * @param {string} data.giftId - the uid for the gift
+   * @param {bool} data.isSent - if the gift is sent
    * @returns {Result}
    */
-  async ({ user, value, gifteeGameAccountUUID }, context) => {
-    // this has to mark both the giftee and gifter
-
-    if (!gifteeGameAccountUUID) {
-      return { error: "no gifteeGameAccountUUID set" };
-    }
-
-    if (typeof value === "undefined") {
-      value = true;
-    }
-    // gifter first
+  async ({ user, giftId, isSent }) => {
     let gameAccountUUID = await getGameAccountUUID(user);
     if (gameAccountUUID.error) {
       return { error: "no API key set" };
     }
-    let entryResult = await db
-      .collection(CollectionTypes.EVENTS)
-      .doc(EVENT)
-      .collection(CollectionTypes.EVENTS__PARTICIPANTS)
-      .doc(gameAccountUUID.success)
-      .set({ sent_own: value }, { merge: true })
+
+    let eventDoc = db.collection(CollectionTypes.EVENTS).doc(EVENT);
+
+    let giftDoc = eventDoc
+      .collection(CollectionTypes.EVENTS__GIFTS)
+      .doc(giftId);
+
+    if (!giftDoc.exists) {
+      return { error: `Found no gifts with id: ${giftId}` };
+    }
+
+    let isGiftUpdatedSuccessfully = giftDoc
+      .set(
+        {
+          sent: isSent ? new Date().toISOString() : null
+        },
+        { merge: true }
+      )
       .then(() => {
         return true;
       })
       .catch(() => {
         return false;
       });
 
-    // giftee now, the giftee's gameAccountUUID is known
-    let gifteeStatus = await markGifteeAccount(
```

Comment:
part of teh ``markGifteeAccount`` function was to update the global stats

Reply:
This has been resolved now! Thanks for reminding me